libcalico-go/lib/validator/v3/validator_test.go

```diff
@@ -665,6 +667,7 @@ func init() {
 		Entry("should reject : in an interface", libapiv3.WorkloadEndpointSpec{InterfaceName: "Invalid:Intface"}, false),
 
 		// (API) FelixConfiguration.
+		Entry("should accept a valid IptablesBackend value 'Auto'", api.FelixConfigurationSpec{IptablesBackend: &iptablesBackendAuto}, true),
```

Comment:
Could you add cases for accepting "NFT" and "Legacy" and rejecting an invalid value here as well?